src/shared/sam/activation.ts

```diff
@@ -287,6 +288,33 @@ async function createYamlExtensionPrompt(): Promise<void> {
             yamlPromptDisposables
         )
 
+        /**
+         * Prompt the user to install the YAML plugin when AWSTemplateFormatVersion becomes available as a top level key
+         * in the document
+         * @param event An vscode text document change event
+         * @returns nothing
+         */
+        function promptOnAWSTemplateFormatVersion(event: vscode.TextDocumentChangeEvent): void {
+            for (let line = 0; line < event.document.lineCount; line++) {
+                const textLine = event.document.lineAt(line)
+                if (!textLine.text.startsWith(' ') && textLine.text.includes('AWSTemplateFormatVersion')) {
+                    promptInstallYamlPlugin(yamlPromptDisposables)
+                    return
+                }
+            }
+        }
+        const debouncedUpdate = debounce(promptOnAWSTemplateFormatVersion, 1000)
```

Comment:
`debounce` wouldn't be needed if we limited one task per document, something like:
```ts
const notifications = new Map<string, Promise<unknown>>()

// in the event handler
const key = document.uri.toString()
if (!notifications.has(key)) { // and whatever other checks
    notifications.set(key, promptOnAWSTemplateFormatVersion.finally(() => notifications.delete(key)))
}
```

I need to make a generic form of this pattern though as I see it a lot. 

This is overkill but you could also not delete the promise to have it only show at most 1 notification per document but it'd be a memory leak. A weak ref would be needed against the document uri.